crates/router/src/core/payments/operations/payment_update.rs

```diff
@@ -567,3 +565,57 @@ impl<F: Send + Clone> ValidateRequest<F, api::PaymentsRequest> for PaymentUpdate
         ))
     }
 }
+
+impl PaymentUpdate {
+    fn populate_payment_attempt_with_request(
+        payment_attempt: &mut storage::PaymentAttempt,
+        request: &api::PaymentsRequest,
+    ) {
+        payment_attempt.business_sub_label = request
+            .business_sub_label
+            .clone()
+            .or(payment_attempt.business_sub_label.clone());
+        payment_attempt.payment_method_type = request
+            .payment_method_type
+            .map(|pmt| pmt.foreign_into())
+            .or(payment_attempt.payment_method_type.clone());
+        payment_attempt.payment_experience = request
+            .payment_experience
+            .map(|experience| experience.foreign_into());
+        payment_attempt.amount_to_capture = request
+            .amount_to_capture
+            .or(payment_attempt.amount_to_capture);
+    }
+    fn populate_payment_intent_with_request(
+        payment_intent: &mut storage::PaymentIntent,
+        request: &api::PaymentsRequest,
+    ) {
+        payment_intent.return_url = request.return_url.as_ref().map(|a| a.to_string());
+
+        payment_intent.business_country = request
+            .business_country
+            .unwrap_or(payment_intent.business_country);
+        payment_intent.business_label = request
+            .business_label
+            .clone()
+            .unwrap_or(payment_intent.business_label.clone());
+        payment_intent.description = request
+            .description
+            .clone()
+            .or_else(|| payment_intent.description.clone());
+        payment_intent.statement_descriptor_name = request
+            .statement_descriptor_name
+            .clone()
+            .or_else(|| payment_intent.statement_descriptor_name.clone());
+
+        payment_intent.statement_descriptor_suffix = request
+            .statement_descriptor_suffix
+            .clone()
+            .or_else(|| payment_intent.statement_descriptor_suffix.clone());
```

Comment:
```suggestion
        request.statement_descriptor_suffix
            .map(|i| payment_intent.statement_descriptor_suffix.replace(i.clone()))
```

could avoid more clones here, Although i think this is anyways optimized away...